File: Backend/Models/Metrics/MetricType.cs
using Helse.Models.Common;

namespace Helse.Models.Metrics;

public class UpdateMetricType : CreateMetricType
{
    public required long Id { get; set; }
}

public class CreateMetricType : MetricTypeBase
{
    public required int Unit { get; set; }
}

public class MetricType : MetricTypeBase
{
    public required long Id { get; set; }

    public required Unit Unit { get; set; }
}

public class MetricTypeBase
{
    public required string Name { get; set; }

    public MetricSummary SummaryType { get; set; }

    public string? Description { get; set; }

    public MetricDataType Type { get; set; }

    public required bool UserEditable { get; set; }

    public bool Visible { get; set; }

    public bool ShowOnDashboard { get; set; }

    public required long GroupId { get; set; }

    public long? ValueCount { get; set; }

    public TimeSpan? TimeDifference { get; set; }
}

File: flutter_frontend/lib/services/local/database/models/metric_type.dart
import 'package:drift/drift.dart';
import 'package:helse/services/local/database/models/helse_table.dart';
import 'package:helse/services/local/database/models/metric.dart';
import 'package:helse/services/local/database/models/unit.dart';

class MetricType extends HelseTable {
  TextColumn get name => text()();
  TextColumn get description => text().nullable()();
  late final groupId = integer().references(Group, #id)();
  TextColumn get timeDifference => text().nullable()();
  BoolColumn get visible => boolean()();
  IntColumn get valueCount => integer().nullable()();
  BoolColumn get showOnDashboard => boolean()();
  BoolColumn get userEditable => boolean()();
  TextColumn get summaryType => text()();
  TextColumn get type => text()();
  late final unit = integer().references(Unit, #id)();
}
